pkg/image/registry/imagestream/etcd/etcd.go

```diff
@@ -91,6 +91,15 @@ func (r *REST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object, boo
 	return r.store.Update(ctx, obj)
 }
 
+// UpdateAll updates both the image stream specification and status.
+func (r *REST) UpdateAll(ctx kapi.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	oldStrategy := r.store.UpdateStrategy
+	r.store.UpdateStrategy = imagestream.NewAllStrategy(oldStrategy.(imagestream.Strategy))
```

Comment:
You can't do this - it's illegal to change anything in `r` during an edit.  You have to have a separate object to manage that.